### PR TITLE
Add HACKING.md with hints for working on Eio

### PR DESCRIPTION
**Pull request by talex5**

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+```
+git clone --recursive https://github.com/ocaml-multicore/eio.git
+cd eio
+opam pin -yn .
+opam install eio_main
+```
+
+The `--recursive` flag is useful because we sometimes use Git submodules
+to vendor libraries that haven't been released yet.
```
[the remark this answers is not in the document]
Comment (talex5) [reply HACKING.md:13]: Mmm, probably. I added it in anticipation of the https://github.com/talex5/eio/tree/eio-posix backend getting submitted soon, which currently vendors @haesbaert's `iomux` library. But it's probably better to use a subtree, or just release iomux to opam-repository before merging that branch.

File: HACKING.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+### Cross-platform unit tests
+
+These are in the top-level `tests` directory.
+They are run against whichever backend `Eio_main.run` selects, and therefore must get the same result for all backends.
+
+### Concurrency primitives
+
+`lib_eio/tests` tests some internal data structures, such as the lock-free cells abstraction.
+The `.md` files in that directory provide a simple walk-through to demonstrate the basic operation,
+while `lib_eio/tests/dscheck` uses [dscheck][] to perform exhaustive testing of all atomic interleavings
```
Comment (polytypic) [inline HACKING.md:51]: Broken link here?

Comment (talex5) [reply HACKING.md:51]: It works for me.

Comment (talex5) [reply HACKING.md:51]: I'm using the preview at https://github.com/ocaml-multicore/eio/blob/e332fdbfec500f7ac6f4380f32278f734053990c/HACKING.md#concurrency-primitives - does that work for you?

Comment (polytypic) [reply HACKING.md:51]: Yes, it does.   I didn't know about reference-style links &mdash; and now that I do, I notice they are actually documented on a page I'm pretty sure I've read before on markdown!  What is this magic, a heisenfeature?  😅

Carry on!  LGTM

**Comment by polytypic** [comment]
One thing I'd like to suggest is to use e.g. [prettier](https://prettier.io/) (but any tool that is relatively easy to integrate with various editors or can be easily obtained and invoked from the command line is fine with me) to automatically format the markdown files.

File: HACKING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (talex5) [reply HACKING.md]: I've removed it.

**Comment by talex5** [comment]
> One thing I'd like to suggest is to use e.g. [prettier](https://prettier.io/) (but any tool that is relatively easy to integrate with various editors or can be easily obtained and invoked from the command line is fine with me) to automatically format the markdown files.

I pasted this file into the online test page, but it didn't seem to make any difference. I guess people can just run formatters locally if they want help while editing?

**Pull request merged**